modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -354,6 +395,69 @@ struct IEUnit {
     }
 };
 
+bool IEUnit::InputFrameDesc::is_applicable(const cv::GMetaArg &mm) {
+    switch (mm.index()) {
+        case cv::GMetaArg::index_of<cv::GFrameDesc>():
+            return true;
+        default:
+            return false;
+    }
+}
+
+const IEUnit::InputFrameDesc::description_type &
+IEUnit::InputFrameDesc::get_param(const input_name_type &input) const {
+    auto it = map.find(input);
+    GAPI_Assert(it != map.end() && "No appropriate input is found in InputFrameDesc");
+    return it->second;
+}
+
+void IEUnit::InputFrameDesc::set_param(const input_name_type &input,
+                                       const IE::InputInfo::Ptr& ii) {
+    GAPI_DbgAssert(ii && "IE::InputInfo::Ptr is nullptr");
+    description_type ret;
+    ret.fmt = cv::MediaFormat::NV12;
+    const InferenceEngine::SizeVector& inDims = ii->getTensorDesc().getDims();
+    auto layout = ii->getTensorDesc().getLayout();
+    GAPI_LOG_DEBUG(nullptr, "network input: " << ii->name() <<
+                            ", tensor dims: " << inDims[0] << ", " << inDims[1] <<
+                             ", " << inDims[2] << ", " << inDims[3]);
+    if (layout == InferenceEngine::NHWC) {
```

Comment:
thanks, but current code has  more vast covering when it sticks `documented` logic in parsing layout description: my idea is - since `layout` is introduced (and documented) then we must adhere its meaning despite on any verbal considerations about behavior

Reply:
Discussed online